### PR TITLE
(docs): add react props table to component doc pages

### PR DESCRIPTION
**Pull request by vikas199**

### CODE REVIEW

File: libs/www/src/modules/components/modules/Badge/Badge.mdx [hunk 1]
```diff
@@ -12,6 +12,7 @@ Use a `Badge` to enumerate the common classifications of a type of thing.
 
 <BadgePropdefs />
 
+
```
Comment (NagaPradeepGajula-osi) [inline libs/www/src/modules/components/modules/Badge/Badge.mdx]: unnecessary white space. Probably typo.

File: libs/www/src/modules/components/modules/Badge/BadgePropdefs.jsx [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
 import docgenComposed from '../../../../macros/docgen-composed.macro'
 const docgenInfo = docgenComposed('reactstrap/src/Badge.js')
+console.log(JSON.stringify(docgenInfo))
 
-export default () => <PropTable propDefs={docgenInfo} />
+export default () => (
+  <ReactPropTable propMetaData={JSON.stringify(docgenInfo.props, null, 2)} />
+)
+
+//export default () => <PropTable propDefs={docgenInfo} />
```
Comment (NagaPradeepGajula-osi) [inline libs/www/src/modules/components/modules/Badge/BadgePropdefs.jsx]: remove comments.

File: libs/www/src/modules/components/modules/Badge/BadgePropdefs.jsx [hunk 3]
```diff
@@ -1,6 +1,12 @@
 import React from 'react'
 import PropTable from '../../../../components/PropTable'
+import ReactPropTable from '../../../../components/PropTable'
 import docgenComposed from '../../../../macros/docgen-composed.macro'
 const docgenInfo = docgenComposed('reactstrap/src/Badge.js')
+console.log(JSON.stringify(docgenInfo))
```
Comment (NagaPradeepGajula-osi) [inline libs/www/src/modules/components/modules/Badge/BadgePropdefs.jsx]: no console logs please.

**Comment by NagaPradeepGajula-osi** [comment]
But the code's looking great. Can you add the screenshots of after the changes when you have this done?

File: libs/www/src/components/PropDefs/AvatarProps.jsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import { ReactPropTable } from '../ReactPropTable'
+import { propDefsFunc } from '../ReactPropTable'
+import docgenComposed from '../../macros/docgen-composed.macro'
+
+const docgenInfo = docgenComposed('@cwds/components/src/Avatar/Avatar.jsx')
+const ButtonGroupPropdefs = () => (
```
Comment (eckdanny-osi) [inline libs/www/src/components/PropDefs/AvatarProps.jsx]: @vikas199 were we unsuccessful creating a generic `<PropsTable /> | fn(...): PropsTable`? I really don't want a `*Props.jsx` for _every_ component we have cluttering up the fs

File: libs/www/src/modules/components/modules/Avatar/Avatar.mdx [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import { Link } from 'react-router-dom';
 import { Avatar, Alert, Card, Row, Col } from '@cwds/components';
 import DemoCard from '../../../../components/DemoCard';
+import PropTable from '../../../../components/PropDefs/AvatarProps';
```
Comment (eckdanny-osi) [inline libs/www/src/modules/components/modules/Avatar/Avatar.mdx]: What was the outcome of using the `docgen-composed.macro` from within an MDX context?

File: libs/www/src/modules/components/modules/Avatar/AvatarProps.jsx [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import { ReactPropTable } from '../../../../../src/components/ReactPropTable'
+import { propDefsFunc } from '../../../../../src/components/ReactPropTable'
+import docgenComposed from '../../../../macros/docgen-composed.macro'
+
+const docgenInfo = docgenComposed('@cwds/components/src/Avatar/Avatar.jsx')
+const ButtonGroupPropdefs = () => (
```
Comment (eckdanny-osi) [inline libs/www/src/modules/components/modules/Avatar/AvatarProps.jsx]: `ButtonGroupPropdefs`?

File: libs/www/src/components/PropDefs/AvatarProps.jsx (hunk 4 above)
Comment (vikas199) [reply libs/www/src/components/PropDefs/AvatarProps.jsx]: Yes, we are not able to make it in MDX file. so I need to create a Props.jsx for each component till we figure out some alternate way to directly use in the MDX file

**Comment by eckdanny-osi** [comment]
I _really_ don't like creating so many `*PropDef` files 😢. One of the main value propositions of the `docgen-composed.macro` was to provide propdefs at compile-time _without_ having to have a file per component (on disk, less in version ctrl).

The root issue here is (probably?) that the webpack inline-loader syntax is getting confused with handling `babel-macros`. We can explore options (like ejecting CRA or forking `react-scripts` so that we may add an `md[x]` loading chain... 

I'm OK taking on some technical debt for now... Providing consumers of `@cwds/components` (cares developers) with propTables is _definitely_ a value-add move. Lemme pull down the PR locally and attempt some low-effort workarounds before we merge.

**Comment by eckdanny-osi** [comment]
@vikas199 and me chatted. Agree we _reallly_ would like MDX and babel-macros to work together, but there is not sol'n apparant _right now_ that would solve that. We should spike on ejecting/forking react-scripts.

Let's merge

File: libs/www/src/components/ReactPropTable.jsx [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+      defaultValue: prop.defaultValue,
+    }
+  })
+}
+
+export const ReactPropTable = ({ propDefinitions }) => {
+  return (
+    <div>
+      <PropsTable
+        type={() => {}}
```
Comment (eckdanny-osi) [inline libs/www/src/components/ReactPropTable.jsx]: still a mystery?

**Pull request merged**